src/Cool/CollaboraConnection.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+namespace Drupal\collabora_online\Cool;
+
+use Drupal\collabora_online\Exception\CollaboraNotAvailableException;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\RequestOptions;
+use Psr\Http\Client\ClientExceptionInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Service to load the discovery.xml from the Collabora server.
+ */
+class CollaboraConnection {
```

Comment:
Yeah the class name sucks.
I do think it is nice to separate the two parts of fetching the xml vs parsing it and providing its data.
This way we also split the dependencies (guzzle), and modules can replace or decorate one service without touching the other.

For the name, the big question is what else this service will do in the future.
My plan would be that this service (connection or however we call it) fetches files or other from the Collabora server, which in the future might be other things beyond the discovery.xml.
We could also make the interface more generic, e.g. `->fetch($name)` instead of `getDiscoveryXml()`.

Reply:
In core I looked for services/classes that use the http client.
I found several that are named like `SomethingFetcher`.
E.g. ResourceFetcher, UpdateFetcher, SecurityAdvisoriesFetcher.
Some return parsed json, others (at least one) returns the raw response body as string.